libs/partners/huggingface/langchain_huggingface/chat_models/huggingface.py

```diff
@@ -113,6 +127,33 @@ def _convert_TGI_message_to_LC_message(
     return AIMessage(content=content, additional_kwargs=additional_kwargs)
 
 
+def _convert_hf_kwargs_llm_to_chat_completions(
+    **kwargs: Any,
+) -> Any:
+    """
+    Convert a set of keyword arguments to be compatible with [InferenceClient.chat_completion](https://huggingface.co/docs/huggingface_hub/v0.26.2/en/package_reference/inference_client#huggingface_hub.InferenceClient.chat_completion)
+    Overrides mapped keyword arguments and omits incompatible keyword arguments.
+    """
+    from huggingface_hub import InferenceClient  # type: ignore[import-untyped]
+
+    for k, v in LLM_INFERENCE_CLIENT_MAP.items():
+        if k in kwargs:
+            kwargs[v] = kwargs.pop(k)
+
+    sig = set(signature(InferenceClient.chat_completion).parameters)
```

Comment:
could you move this to init instead and do additional validation:

(1) if the user provides `HuggingFaceEndpoint` as an LLM, check if `huggingface_hub` is installed, otherwise raise an informative error like here https://github.com/langchain-ai/langchain/blob/master/libs/partners/huggingface/langchain_huggingface/llms/huggingface_endpoint.py#L195
(2) then set `hf_endpoint_kwargs` / `hf_supported_kwargs` on the class, in the init. you could also initialize it with the values supplied in the init (like `max_new_tokens`) and use it in `_invocation_params`

Reply:
I do not completely understand 1; If the user passed an `HuggingFaceEndpoint`, then the pydantic validation you linked has already been done (including the availability check of `huggingface_hub`)
```
@model_validator(mode="after")
    def validate_environment(self) -> Self:
```

Or am I missing something?